pkg/controller/v1alpha1/inferencegraph/raw_ig.go

```diff
@@ -180,6 +211,144 @@ func handleInferenceGraphRawDeployment(cl client.Client, clientset kubernetes.In
 	return deployment[0], reconciler.URL, nil
 }
 
+func handleInferenceGraphRawAuthResources(ctx context.Context, clientset kubernetes.Interface, scheme *runtime.Scheme, graph *v1alpha1api.InferenceGraph) error {
+	saName := getServiceAccountNameForGraph(graph)
+
+	if graph.GetAnnotations()[constants.ODHKserveRawAuth] == "true" {
+		graphGVK, err := apiutil.GVKForObject(graph, scheme)
+		if err != nil {
+			return errors.Wrapf(err, "fails get GVK for inference graph")
+		}
+		ownerReference := metav1cfg.OwnerReference().
+			WithKind(graphGVK.Kind).
+			WithAPIVersion(graphGVK.GroupVersion().String()).
+			WithName(graph.GetName()).
+			WithUID(graph.UID).
+			WithBlockOwnerDeletion(true).
+			WithController(true)
+
+		// Create a Service Account that can be used to check auth
+		saAuthVerifier := corev1cfg.ServiceAccount(saName, graph.GetNamespace()).
+			WithOwnerReferences(ownerReference)
+		_, err = clientset.CoreV1().ServiceAccounts(graph.GetNamespace()).Apply(ctx, saAuthVerifier, metav1.ApplyOptions{FieldManager: InferenceGraphControllerName})
+		if err != nil {
+			return errors.Wrapf(err, "fails to apply auth-verifier service account for inference graph")
+		}
+
+		// Bind the required privileges to the Service Account
+		err = addAuthPrivilegesToGraphServiceAccount(ctx, clientset, graph)
```

Comment:
nit:
```suggestion
		err = addAuthPrivilegesToGraphServiceAccount(ctx, clientset, graph, saName)
```

Reply:
So, do you prefer sending it as an argument rather than re-getting it inside the func?

Reply:
yes, would avoid having to re-get. but it's only a nit :)